File: algorithms/implementation/library-fine/library-fine-javascript/js/test_libraryFine.js
import {libraryFine} from '../js/libraryFine.js';
import assert from 'assert';

describe('libraryFine', function() {

    it('O valor da multa é: 45', function() {

        let d1 = 9
        let m1 = 6
        let a1 = 2015
        
        let d2 = 6
        let m2 = 6
        let a2 = 2015
        
        let esperado = 45
        
        let resultado =  libraryFine(d1, m1, a1, d2, m2, a2)

        assert.strictEqual(resultado, esperado);
    });
});

describe('libraryFine', function() {

    it('O valor da multa é: 0', function() {

        let d1 = 6;
        let m1 = 6;
        let a1 = 2015;
        
        let d2 = 9;
        let m2 = 6;
        let a2 = 2016;
        
        let esperado = 0
        
        let resultado =  libraryFine(d1, m1, a1, d2, m2, a2)

        assert.strictEqual(resultado, esperado);
    });
});

describe('libraryFine', function() {

    it('O valor da multa é: 10000', function() {

        let d1 = 2;
        let m1 = 7;
        let a1 = 2015;
        
        let d2 = 1;
        let m2 = 2;
        let a2 = 2014;
        
        let esperado = 10000;
        
        let resultado =  libraryFine(d1, m1, a1, d2, m2, a2)

        assert.strictEqual(resultado, esperado);
    });
});

File: algorithms/implementation/library-fine/library-fine-python/tests/test_libraryFine.py
from libraryFine import libraryFine

class TesteLibraryFine:
    
    def testeCaso1(self):
        
        d1 = 9
        m1 = 6
        a1 = 2015
        
        d2 = 6
        m2 = 6
        a2 = 2015
        
        esperado = 45
        
        resultado =  libraryFine(d1, m1, a1, d2, m2, a2)
        
        assert resultado == esperado
        
        
    def testeCaso2(self):
        
        d1 = 6
        m1 = 6
        a1 = 2015
        
        d2 = 9
        m2 = 6
        a2 = 2016
        
        esperado = 0
        
        resultado =  libraryFine(d1, m1, a1, d2, m2, a2)
        
        assert resultado == esperado
        
    def testeCaso3(self):
        
        d1 = 2
        m1 = 7
        a1 = 2015
        
        d2 = 1
        m2 = 2
        a2 = 2014
        
        esperado = 10000
        
        resultado =  libraryFine(d1, m1, a1, d2, m2, a2)
        
        assert resultado == esperado